Assets/Packages/com.studio23.ss2.puzzlesystem/Runtime/Script/Model/PuzzleInfo.cs

```diff
@@ -35,17 +35,29 @@ public class PuzzleInfo
         public List<int> ResultValues { get; set; }
         
         
-        
 
         /// <summary>
         /// Gets or sets a value indicating whether the puzzle is solved.
         /// </summary>
-        public bool IsPuzzleSolved { get; set; }
+        public bool IsPuzzleSolved  {
+            get
+            {
+                var isPuzzleUnlock = true;
+                for (int i = 0; i < ResultValues.Count; i++)
+                {
+                    if (ResultValues[i] != CurrentValues[i])
+                    {
+                        isPuzzleUnlock = false;
+                        break;
+                    }
+                }
 
-        /// <summary>
-        /// Event invoked when the puzzle is unlocked.
-        /// </summary>
-        public Action OnPuzzleSolved;
+                return isPuzzleUnlock;
+            }
+            private set { throw new NotImplementedException(); }
```

Comment:
Must implement before merging to main. Can't have throwing exceptions in the published repo